app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -162,6 +165,27 @@ class PersonalDetailsCollectionController @Inject()(personalDetailsSubmission: P
     }
   }
 
+  def showHaveYourNationalInsuranceNumber(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
+    viewConfig.isLoggedIn.flatMap { isLoggedIn: Boolean =>
+      Future.successful(Ok(do_you_have_your_nino(DoYouHaveYourNino.apply(), completionUrl, isLoggedIn, failureUrl)))
+    }
+  }
+
+  def processHaveYourNationalInsuranceNumber(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
This method could do with reformatting for readability - long lines and confusing brackets on the fold